test/integration/cni/pod_egress_traffic_test.go

```diff
@@ -0,0 +1,286 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cni
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+	coreV1 "k8s.io/api/core/v1"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/integration/common"
+)
+
+const (
+	cmdForPodEgressIpv6 = `ip -f inet6 address show dev v6if0 | sed -n "s/.*inet6 \(fd00::ac:.*\)\/118.*/\1/p"`
+	cmdForPodEgressIpv4 = `ip -f inet address show dev v4if0 | sed -n "s/.*inet \(169.254..*\)\/22.*/\1/p"`
+)
+
+// Verifies connectivity to deployment behind different service types
+var _ = Describe("[CANARY] test cluster egress connectivity", func() {
+	var (
+		err error
+		// test container that verifies connectivity to en external IPv6 using IPv6 only
+		testerContainer coreV1.Container
+
+		// Primary node busybox deployment
+		primaryNodeDeployment *v1.Deployment
+		// Secondary node busybox deployment
+		secondaryNodeDeployment *v1.Deployment
+
+		primaryNodeIP   *string
+		secondaryNodeIP *string
+
+		isIPv4Cluster bool
+
+		// pods in running primary node
+		podsInPrimaryNode coreV1.PodList
+		// pods in running secondary node
+		podsInSecondaryNode coreV1.PodList
+		// IPs of PODs running in primary node
+		podsInPrimaryNodeIPs []string
+		// IPs of PODs running in secondary node
+		podsInSecondaryNodeIPs []string
+	)
+
+	BeforeEach(func() {
+		// initialize vars
+		err = nil
+		primaryNodeIP = nil
+		secondaryNodeIP = nil
+		testerContainer = coreV1.Container{}
+
+		// determine v4 or v6 cluster
+		clusterOutput, err := f.CloudServices.EKS().DescribeCluster(f.Options.ClusterName)
+		Expect(err).NotTo(HaveOccurred())
+		if *clusterOutput.Cluster.KubernetesNetworkConfig.IpFamily != "ipv6" {
+			isIPv4Cluster = true
+		}
+
+		// determine primary and secondary nodes IPs
```

Comment:
The BeforSuite is used for all test specs under `cni` folder, and our primayNodeIP and secondaryNodeIP  have special meaning in our egress test -- it is the SNAT address used by PODs egress and it may not be what other test specs expects.